packages/app/src/components-styled/legenda.tsx

```diff
@@ -2,24 +2,37 @@ import css from '@styled-system/css';
 import * as React from 'react';
 import styled from 'styled-components';
 
-export type LegendShape = 'line' | 'square' | 'circle';
+export type LegendShape = 'line' | 'square' | 'circle' | 'custom';
 
-export type LegendItem = { color: string; label: string; shape: LegendShape };
+export type LegendItem = {
+  color: string;
+  label: string;
+  shape: LegendShape;
+  ShapeComponent?: any;
```

Comment:
Prettier imho:

```suggestion
  shapeComponent?: ReactNode;
```

and in the calling context you can pass the shape with `shapeComponent: <HatchedSquare />`

Reply:
tried it, got into a fight with the compiler, lost my patience